### DISCUSSION TITLE
frame around the plot

### DISCUSSION THREAD

**Discussion by ShubhamBhut** [shap/shap#3010] [Q&A] [answered]
Hi, I am planning to customize my plots generated by SHAP. And I came to know that shap doesn't support all the matplotlib features. primarily considering - `plt.figure()`. I actually want to set a frame around my plot (done by edgecolor and edgewidth in matplotlib). Can anyone tell if there is any other way to set frame ?

**Comment by thatlittleboy** [comment]
@ShubhamBhut The plot functions come with a show=False option, in which case you'll get the figure object back.

Then you can make whatever modifications you want before doing plt.show().

**Reply by ShubhamBhut** [reply]
I have used show=False but still plt.figure didn't work for me. My guess is that shap_plots already initiates a plt.figure for itself, I haven't read the source code so not sure.

**Reply by thatlittleboy** [reply]
yeah you're not supposed to initiate another `plt.figure()` again. Just use the Figure that the plot functions return you and build on top of that.

Can you show the code that you used?

**Reply by ShubhamBhut** [reply]
Actually that code is in my office machine I can't access it right now, but I got what you said. I will check it and response you when I have access.

**Comment by ShubhamBhut** [comment]
@thatlittleboy 
The below code runs fine
```
plt.figure(linewidth=5, edgecolor="#04253a")
shap.summary_plot(shap_values, show=False)
plt.xticks(fontsize=10, fontweight='bold')
plt.yticks(fontsize=12, fontweight='bold')
plt.ylabel.fontweight = 'bold'
plt.title("SHAP value variation", fontdict={'fontweight':'bold'})
plt.show()
```

However the below code which seems exactly seems doesn't add frame. Basically whole `plt.figure `doesn't work there

```
import shap
import matplotlib.pyplot as plt

shap_values = shap_values # Replace with your SHAP values

feature_names = ['Rotation Speed (rpm)', 'Traverse speed (mm/min)',
       'Traverse distance (mm)', 'Distance from Shoulder (mm)']  # Replace with your feature names


for i, feature in enumerate(feature_names):
    plt.figure(linewidth=5, edgecolor="#04253a")
    fig = shap.plots.scatter(shap_values[:, i],show=False, dot_size=26)  
    plt.title(f"SHAP Values for {feature}", fontdict={'fontweight': 'bold'})
    plt.xlabel(f"{feature}", fontdict={'fontweight': 'bold'})
    plt.ylabel("SHAP Value", fontdict={'fontweight': 'bold'})
    plt.xticks(fontsize=10, fontweight='bold')
    plt.yticks(fontsize=12, fontweight='bold')
    plt.gca().spines['top'].set_visible(False)
    plt.gca().spines['right'].set_visible(False)
    plt.gca().spines['bottom'].set_linewidth(3)
    plt.gca().spines['left'].set_linewidth(3)

    plt.show()
```

**Reply by thatlittleboy** [reply, marked as answer]
@ShubhamBhut Sorry for the late reply. Was busy at work. I think you have 2 options.

If you're using the scatter plot, it currently accepts an `ax` argument, so you can create the figure and axis first (with the necessary figure linewidth and edgecolor settings) & pass it into `shap.plots.scatter`, or you can `plt.gcf()` after the fact, and modify the figure then.

Option 1:
```python
for i, feature in enumerate(feature_names):
    f = plt.figure(linewidth=5, edgecolor="#04253a")
    ax = f.gca()
    fig = shap.plots.scatter(shap_values[:, i], ax=ax, show=True, dot_size=26)
```

Option 2:
```python
for i, feature in enumerate(feature_names):
    fig = shap.plots.scatter(shap_values[:, i],show=False, dot_size=26)
    f = plt.gcf()
    f.set_linewidth(5)
    f.set_edgecolor("#04253a")
    plt.show()
```

**Reply by ShubhamBhut** [reply]
Thank you very much for your help. This solved the issue.
